source/common/request_source_plugin_impl.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include "envoy/registry/registry.h"
+
+#include "nighthawk/common/request_source_plugin.h"
+
+#include "api/client/options.pb.h"
+#include "api/request_source/request_source_plugin_impl.pb.h"
+
+#include "common/uri_impl.h"
+
+namespace Nighthawk {
+
+/**
+ */
+class DummyRequestSourcePlugin : public RequestSourcePlugin {
+public:
+  explicit DummyRequestSourcePlugin(
+      const nighthawk::request_source::DummyPluginRequestSourceConfig& config,
+      Envoy::Api::Api& api);
+  RequestGenerator get() override;
+  /**
+   * Will be called on an intialized and running worker thread, before commencing actual work.
+   * Can be used to prepare the request source implementation (opening any connection or files
+   * needed, for example).
+   */
+  void initOnThread() override;
+
+private:
+  const double dummy_value_;
+};
+
+/**
+ * Factory that creates a DummyRequestSourcePlugin from a DummyRequestSourcePluginConfig proto.
+ * Registered as an Envoy plugin.
+ */
+class DummyRequestSourceConfigFactory : public virtual RequestSourcePluginConfigFactory {
+public:
+  std::string name() const override;
+  Envoy::ProtobufTypes::MessagePtr createEmptyConfigProto() override;
+  RequestSourcePluginPtr createRequestSourcePlugin(const Envoy::Protobuf::Message& message,
+                                                   Envoy::Api::Api& api) override;
+};
+
+// This factory is activated through ???.
+DECLARE_FACTORY(DummyRequestSourceConfigFactory);
+
+using RequestOptionsIterator =
+    Envoy::ProtobufWkt::internal::RepeatedPtrIterator<const nighthawk::client::RequestOptions>;
+
+/**
+ */
+class FileBasedRequestSourcePlugin : public RequestSourcePlugin {
```

Comment:
If I understand correctly, this plugin will replay request specifications from a file. This file gets read into memory by each worker, and each worker will replay it. Is that correct? Maybe it's worth adding a doc comment at the class level for this?

Reply:
Good point.